### ISSUE TITLE
Multiple redirect URIs

### ISSUE THREAD

**Issue by jerometremblay** [brandnewbox/oidc_provider#17]
I deploy an application in two environments, that each have their own URL.

In my provider, I tried to setup multiple valid redirect_uri like this:

```
OIDCProvider.configure do |config|

  config.add_client do  
    name client_identifier
    identifier client_identifier
    secret client_data[:secret]
    redirect_uri ['https://url1', 'https://url2']
  end
end
```

Unfortunately, when I do that I got the following error in my client application:

`E, [2023-10-19T16:10:26.953023 #1] ERROR -- omniauth: (xxx) Authentication failure! invalid_client: Rack::OAuth2::Client::Error, invalid_client :: The client identifier provided is invalid, the client failed to authenticate, the client did not include its credentials, provided multiple client credentials, or used unsupported credentials type.`

Reverting to a single redirect_uri fixed the issue. But I had to create a new client for each instances of my application, which increased complexity. I thought that by having multiple valid redirect_uri I could use the same client settings in my two environments.

**Comment by willtcarey** [comment]
This was actually just recently addressed in #10. However, I haven't cut a new release of the gem for that, yet.

**Comment by willtcarey** [comment]
@jerometremblay Hey, I think I was mistaken before. We actually had put out the release for this in https://github.com/brandnewbox/oidc_provider/releases/tag/v0.4.0
